src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java

```diff
@@ -0,0 +1,112 @@
+import tracko.logic.commands.order.DeleteOrderCommand;
+import tracko.model.Model;
+import tracko.model.ModelManager;
+import tracko.model.UserPrefs;
+import tracko.model.order.Order;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DeleteCommand}.
+ */
+public class DeleteOrderCommandTest {
+
+    private Model model = new ModelManager(getTrackOWithTypicalOrders(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Order orderToDelete = model.getOrderList().get(INDEX_FIRST_ORDER.getZeroBased());
+        DeleteOrderCommand deleteOrderCommand = new DeleteOrderCommand(INDEX_FIRST_ORDER);
+
+        String expectedMessage = String.format(DeleteOrderCommand.MESSAGE_DELETE_ORDER_SUCCESS, orderToDelete);
+
+        ModelManager expectedModel = new ModelManager(model.getTrackO(), new UserPrefs());
+        expectedModel.deleteOrder(orderToDelete);
+
+        assertCommandSuccess(deleteOrderCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getOrderList().size() + 1);
+        DeleteOrderCommand deleteOrderCommand = new DeleteOrderCommand(outOfBoundIndex);
+
+        assertCommandFailure(deleteOrderCommand, model, Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+    }
+
+    //Test for filtered list, currently not implemented
+    //    @Test
+    //    public void execute_validIndexFilteredList_success() {
+    //        showOrderAtIndex(model, INDEX_FIRST_ORDER);
+    //
+    //        Order orderToDelete = model.getOrderList().get(INDEX_FIRST_ORDER.getZeroBased());
+    //        DeleteOrderCommand deleteOrderCommand = new DeleteOrderCommand(INDEX_FIRST_ORDER);
+    //
+    //        String expectedMessage = String.format(DeleteOrderCommand.MESSAGE_DELETE_ORDER_SUCCESS, orderToDelete);
+    //
+    //        Model expectedModel = new ModelManager(model.getTrackO(), new UserPrefs());
+    //        expectedModel.deleteOrder(orderToDelete);
+    //        showNoPerson(expectedModel);
+    //
+    //        assertCommandSuccess(deleteOrderCommand, model, expectedMessage, expectedModel);
+    //    }
+
+    //Test for filtered list, currently not implemented
+    //    @Test
+    //    public void execute_invalidIndexFilteredList_throwsCommandException() {
+    //        showOrderAtIndex(model, INDEX_FIRST_ORDER);
+    //
+    //        Index outOfBoundIndex = INDEX_SECOND_ORDER;
+    //        // ensures that outOfBoundIndex is still in bounds of address book list
+    //        assertTrue(outOfBoundIndex.getZeroBased() < model.getTrackO().getOrderList().size());
+    //
+    //        DeleteOrderCommand deleteOrderCommand = new DeleteOrderCommand(outOfBoundIndex);
+    //
+    //        assertCommandFailure(deleteOrderCommand, model, Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+    //    }
```

Comment:
I feel that perhaps we should delete this since commented code might result in a messier code base. However, I understand that leaving this here provides easier edits in the future so I'll just raise this as a suggestion.

Reply:
I was thinking about this too but currently since we are still unfamiliar with the code base I think it would be better to leave it in.